src/vncproxy.ml

```diff
@@ -63,43 +63,41 @@ let exn_to_string = function
 
 let find_console rpc session_id x =
   (* Treat 'x' first as a uuid and then as a name_label *)
-  let%lwt vm =
-    try%lwt
-      VM.get_by_uuid rpc session_id x
-    with _ ->
-      let%lwt possibilities = VM.get_by_name_label rpc session_id x in
-      match possibilities with
-      | [ exactly_one ] -> return exactly_one
-      | [] ->
-        Printf.fprintf stderr "Failed to find VM %s\n%!" x;
-        fail (Failure (Printf.sprintf "Failed to find VM %s" x))
-      | x :: too_many ->
-        let%lwt uuid = VM.get_uuid rpc session_id x in
-        Printf.fprintf stderr "More than one VM found with this name.\n%!";
-        Printf.fprintf stderr "Choosing VM with uuid %s.\n%!" uuid;
-        return x in
+  Lwt.catch
+    (fun () -> VM.get_by_uuid ~rpc ~session_id ~uuid:x )
+    (fun  _ ->
+       VM.get_by_name_label ~rpc ~session_id ~label:x >>= function
+       | [ exactly_one ] -> return exactly_one
+       | [] ->
+         Printf.fprintf stderr "Failed to find VM %s\n%!" x;
+         fail (Failure (Printf.sprintf "Failed to find VM %s" x))
+       | y :: _ ->
+         VM.get_uuid ~rpc ~session_id ~self:y >>= fun uuid ->
+         Printf.fprintf stderr "More than one VM found with this name.\n%!";
+         Printf.fprintf stderr "Choosing VM with uuid %s.\n%!" uuid;
+         return y
+
+    ) >>= fun vm ->
   (* Check that the VM is running *)
-  let%lwt power_state = VM.get_power_state rpc session_id vm in
-  let%lwt () =
-    if power_state <> `Running then begin
+  VM.get_power_state ~rpc ~session_id ~self:vm >>= fun power_state ->
+  (if power_state <> `Running then begin
       Printf.fprintf stderr "The VM %s is not running.\n%!" vm;
       fail (Failure (Printf.sprintf "The VM is not running"))
-    end else return () in
+    end else return ()) >>= fun () ->
+
   (* Find a console with the RFB protocol *)
-  let%lwt console =
-    let%lwt all = VM.get_consoles rpc session_id vm in
-    let%lwt all = Lwt_list.map_s (fun c -> Console.get_record rpc session_id c) all in
-    match List.filter (fun c -> c.API.console_protocol = `rfb) all with
-    | [ exactly_one ] -> return exactly_one
-    | [] ->
-      Printf.fprintf stderr "The VM is exposing no VNC consoles.\n%!";
-      fail (Failure (Printf.sprintf "The VM is exposing no VNC consoles"))
-    | x :: too_many ->
-      Printf.fprintf stderr "The VM is exposing multiple VNC consoles.\n";
-      Printf.fprintf stderr "I will choose one for you.\n%!";
-      return x
-  in
-  return console
+  VM.get_consoles ~rpc ~session_id ~self:vm >>= fun all ->
+  Lwt_list.map_s (fun c -> Console.get_record ~rpc ~session_id ~self:c) all >>= fun all ->
```

Comment:
I believe here you can use currying: 
```ocaml
VM.get_consoles ~rpc ~session_id ~self:vm
>>= Lwt_list.map_s (fun c -> Console.get_record ~rpc ~session_id ~self:c)
>>= fun all ->
```
Can you check if it's the case?

Reply:
Ok, it does compile.